From 1fb53b6fb639f45d2ba4bf2a885e3fd5fbe95d8c Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 13 Jun 2021 23:53:26 -0500
Subject: [PATCH 023/124] ASoC: sun4i-spdif: Add support for the D1 variant

The D1 variant is similar to the H6 variant, except for its clock setup.
The clock tree changes impact some register fields on the RX side, but
those are not yet relevant, because RX is not supported by this driver.

Signed-off-by: Samuel Holland <samuel@sholland.org>
---
 sound/soc/sunxi/sun4i-spdif.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sound/soc/sunxi/sun4i-spdif.c b/sound/soc/sunxi/sun4i-spdif.c
index 748fb048a1f8..7c6d596da84f 100644
--- a/sound/soc/sunxi/sun4i-spdif.c
+++ b/sound/soc/sunxi/sun4i-spdif.c
@@ -442,6 +442,12 @@ static const struct sun4i_spdif_quirks sun8i_h3_spdif_quirks = {
 	.val_fctl_ftx   = SUN4I_SPDIF_FCTL_FTX,
 };
 
+static const struct sun4i_spdif_quirks sun20i_d1_spdif_quirks = {
+	.tx_clk_name	= "tx",
+	.reg_dac_txdata = SUN8I_SPDIF_TXFIFO,
+	.val_fctl_ftx   = SUN50I_H6_SPDIF_FCTL_FTX,
+};
+
 static const struct sun4i_spdif_quirks sun50i_h6_spdif_quirks = {
 	.tx_clk_name	= "spdif",
 	.reg_dac_txdata = SUN8I_SPDIF_TXFIFO,
@@ -461,6 +467,10 @@ static const struct of_device_id sun4i_spdif_of_match[] = {
 		.compatible = "allwinner,sun8i-h3-spdif",
 		.data = &sun8i_h3_spdif_quirks,
 	},
+	{
+		.compatible = "allwinner,sun20i-d1-spdif",
+		.data = &sun20i_d1_spdif_quirks,
+	},
 	{
 		.compatible = "allwinner,sun50i-h6-spdif",
 		.data = &sun50i_h6_spdif_quirks,
-- 
2.20.1

